Library/Formula/ocilib.rb

```diff
@@ -0,0 +1,47 @@
+class OracleHomeVar < Requirement
+  fatal true
+  satisfy ENV["ORACLE_HOME"]
+
+  def message; <<-EOS.undent
+      To use this formula you have to set the ORACLE_HOME environment variable.
+      You might need to create a symbolic link without version number
+      for the libclntsh library.
+      Check Oracle Instant Client documentation for more information.
+    EOS
+  end
+end
+
+class Ocilib < Formula
+  desc "Open source C and C++ library for accessing Oracle Databases"
+  homepage "http://www.ocilib.net/"
+  url "https://github.com/vrogier/ocilib/releases/download/v4.1.0/ocilib-4.1.0-gnu.tar.gz"
+  sha256 "d86c1e6d5ef33541299996b76497a1271498cdcf5720ee0f1248a0b78dfb3a0c"
+
+  head "https://github.com/vrogier/ocilib.git"
+
+  option "with-charset-wide", "Use wide charset with Oracle OCI libraries"
+
+  depends_on OracleHomeVar
```

Comment:
I understand and second the pain of installing the Oracle component.

Starting my own tap is probably too much trouble (not being an Oracle DB user). I've been looking into installing the Oracle client as a resource instead, but downloading it from the canonical site requires to be logged in and the license be accepted before each download. If anyone knows about an alternative way to get that component, maybe it'd be a workable alternative (also for `qt5`).